### DISCUSSION TITLE
What is the best way to validate a Renovate config?

### DISCUSSION THREAD

**Discussion by Jacksondr5** [renovatebot/renovate#30458] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I have been running the renovate-config-validator on my PRs any time I change the renovate.json file.  Recently, that started failing asking me to migrate `matchPackagePatterns` to `matchPackageNames`.  But, doing so changed the behavior of my grouping.  Based on [this comment](https://github.com/renovatebot/renovate/discussions/30411#discussioncomment-10165853), I think that's because v38 hasn't rolled out to my org yet.

In doing my research on this issue, I discovered that Renovate can annotate `renovate/reconfigure` PRs ([docs](https://docs.renovatebot.com/config-validation/#validation-of-renovate-config-change-prs)).  I tried that out on a repo that renovate-config-validator was asking me to migrate and it validated successfully:
![image](https://github.com/user-attachments/assets/3b0db553-5d21-4025-9239-8d8a647f684f)

I'm guessing this is because the Renovate app's validation is running on v37 while the renovate-config-validator I'm using is the latest, through `pnpm dlx` (and thus v38).  Is this correct?  If that is correct, would it be better for me to validate my config using the `renovate/reconfigure` branch instead of renovate-config-validator?  I don't think I want to try to pin renovate-config-validator to the version that mend.io is running, the likelihood that I have a mismatch at some point is very high.

**Comment by rarkins** [comment, marked as answer]
Yes, if using the app then using the reconfigure branch would ensure you're tested against the same version as the app runs
